src/commands/deploy.ts

```diff
@@ -214,39 +212,82 @@ async function verifyRuntimeIsCompatible(localRuntime: ProjectRuntime, ui: IAzur
     }
 }
 
-async function tryPublishCSharpProject(deployFsPath: string, outputChannel: vscode.OutputChannel, telemetryProperties: TelemetryProperties): Promise<void> {
+async function runPreDeployTask(deployFsPath: string, telemetryProperties: TelemetryProperties, language: ProjectLanguage, isZipDeploy: boolean): Promise<void> {
+    let taskName: string | undefined = getFuncExtensionSetting(preDeployTaskSetting, deployFsPath);
+    if (!isZipDeploy) {
+        // We don't run pre deploy tasks for non-zipdeploy since that stuff should be handled by kudu
+
+        if (taskName) {
+            // We only need to warn if they have the setting defined
+            ext.outputChannel.appendLine(localize('ignoringPreDeployTask', 'WARNING: Ignoring preDeployTask "{0}" for non-zip deploy.', taskName));
+        }
+
+        return;
+    }
+
+    let isTaskNameDefined: boolean = true;
```

Comment:
why not let isTaskNameDefined = !!taskName;
?

Reply:
Honestly I don't know what to call this variable. It's really "is task name defined is VS Code settings" or "is task name originally defined"

I can't do `!!taskName` because it will always be defined by the time I end up using `isTaskNameDefined`

Reply:
How about isTaskNameDefinedInSettings?

Not that it matters, but that doesn't make sense to me.  Your logic sets isTaskNameDefined=false if (!taskName).  That's equivalent to isTaskNameDefined = !!taskName, meaning if taskName is not undefined and not empty string.

Reply:
Ohhh yes I could do that. I thought you were saying get rid of the `isTaskNameDefined` variable and just do `!!taskName` later on. I get you now